src/main/java/de/osbag/example/repository/UserBaseRepository.java

```diff
@@ -0,0 +1,42 @@
+package de.osbag.example.repository;
+
+import de.osbag.example.controller.UsersController;
+import de.osbag.example.entity.User;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+/**
+ * Created by bijan.zohouri on 4/24/2018.
+ *
+ */
+@Repository
+public class UserBaseRepository {
```

Comment:
Why do we need this class in between ?

Reply:
It is a good practice (prior to JPARepository Interface release) to have your DAO interface as well as its related implementation.(Interface design pattern).  
In order to implement JPARepository I must have implemented all the methods from JPARepository and based on the requirement I didnt need it (we needed save and find) so I used proxy over JPARepository to filter unrelated methods which I didn't want to have them.

Reply:
Having Said that, I just noticed that your point is correct based on the simple requirement which we have had, we might not require the UserBaseRepository anymore. Indeed, we could have put all into the service layer. As I mentioned in the previous comment, in my idea, having the repository in a separate class (even the UserJpaRepository class also can be removed) to isolate your business logic from DAO is a good programming practice.